src/configure/configure.ts

```diff
@@ -432,16 +432,18 @@ class PipelineConfigurer {
 
     private async getAzureResourceDetails(): Promise<void> {
         // show available subscriptions and get the chosen one
-        let subscriptionList = extensionVariables.azureAccountExtensionApi.filters.map((subscriptionObject) => {
+        let subscriptionList = extensionVariables.azureAccountExtensionApi.subscriptions.map((subscriptionObject) => {
             return <QuickPickItemWithData>{
-                label: <string>subscriptionObject.subscription.displayName,
+                label: `${<string>subscriptionObject.subscription.displayName} (${<string>subscriptionObject.subscription.subscriptionId})`,
                 data: subscriptionObject
             };
         });
         let selectedSubscription: QuickPickItemWithData = await this.controlProvider.showQuickPick(constants.SelectSubscription, subscriptionList, { placeHolder: Messages.selectSubscription });
         this.inputs.targetResource.subscriptionId = selectedSubscription.data.subscription.subscriptionId;
+        this.inputs.azureSession = getSubscriptionSession(this.inputs.targetResource.subscriptionId).session;
```

Comment:
Rather than initializing it at multiple places should we do it as part of some initialization.

Reply:
I cannot find a good way to put these two lines in a separate method - it looks too forced. Hence I will leave it this way only.